openshift_pool/openshift/stack.py

```diff
@@ -25,16 +26,20 @@ def __init__(self):
         Loggable.__init__(self)
 
     @cached_property
-    def config_data(self):
+    def openstack_details(self):
         return CONFIG_DATA['openstack']
 
+    @cached_property
+    def ssh_details(self):
+        return CONFIG_DATA['ssh']
+
     @cached_property
     def keystone_client(self):
         return ksclient.Client(
-            username=self.config_data['username'],
-            password=self.config_data['password'],
-            auth_url=self.config_data['auth_url'],
-            tenant_name=self.config_data['tenant_name']
+            username=self.openstack_details['username'],
+            password=self.openstack_details['password'],
+            auth_url=self.openstack_details['auth_url'],
+            tenant_name=self.openstack_details['tenant_name']
```

Comment:
Lets put this in a try/except block. Catch the attribute error and add a relevant log message.

Reply:
@jawatts This attribute MUST be available in the config file so an exception is required here.
I will add a try/except block here and raise a configuration missing exception.